### PR TITLE
fix #275361: Hidden dynamic symbols become visible 2.X->3.0

### PR DESCRIPTION
**Pull request by handrok**
Error was that we need always call element::layout() method. Otherwise position of elements are (0.0 0.0). And painter doesn't draw elements with this position.

### CODE REVIEW

**Comment by anatoly-os** [comment]
And now layout() is called for both visible and invisible elements, but invisible ones don't participate in autoplacement operations, so there is neither performance impact nor invisible elements autoplacement.
This leads to invisible elements collision if you switch visibility, but it is expected I believe, since we position elements for printing, but invisible elements are for playback.

**Comment by handrok** [comment]
@anatoly-os done

File: mscore/musescore.cpp [hunk 1]
```diff
@@ -5502,8 +5502,11 @@ void MuseScore::cmd(QAction* a, const QString& cmd)
             setPlayRepeats(a->isChecked());
       else if (cmd == "pan")
             MScore::panPlayback = !MScore::panPlayback;
-      else if (cmd == "show-invisible")
+      else if (cmd == "show-invisible") {
             cs->setShowInvisible(a->isChecked());
+            cs->setLayoutAll();
```
Comment (wschweer) [inline mscore/musescore.cpp]: show-invisible does not change any layout, so setLayoutAll() should not be called

Comment (handrok) [reply mscore/musescore.cpp]: I added this code because I need it to prevent another bug. When I disabled 'show invisible' and changed some note it led to layout. Layouting led to rebuilding bspTree. And BspTree didn't include such kind of elements (invisible and with disabled option 'show invisible'). When later I enabled 'show Invisible' unvisible elements didn't appear. Because it weren't included to BspTree. To see it there is some changing of score is needed. Because it again leads to rebuild BspTree.

Comment (handrok) [reply mscore/musescore.cpp]: I'll tinking about solution without adding cs->setLayoutAll();

Comment (handrok) [reply mscore/musescore.cpp]: done

File: libmscore/page.cpp [hunk 2]
```diff
@@ -273,7 +273,7 @@ void Page::doRebuildBspTree()
             r = abbox();
 
       bspTree.initialize(r, n);
-      scanElements(&bspTree, &bspInsert, false);
+      scanElements(&bspTree, &bspInsert, true);
```
Comment (mattmcclinch) [inline libmscore/page.cpp]: Why this change? It is the reason for [Hidden Staves appear when Hide Empty Staves enabled](https://musescore.org/en/node/276537).

Comment (anatoly-os) [reply libmscore/page.cpp]: If we set `true` here, hidden elements will be taken into account when building the tree which is not expected. It makes the mentioned issue even harder to fix.

**Comment by mattmcclinch** [comment]
Well, let's set it back to false, if there was no reason for the change. See #4032.

**Comment by anatoly-os** [comment]
There was a reason. I mentioned it here 
https://github.com/musescore/MuseScore/pull/3966#discussion_r224972154. If you specifies `true` here, hidden elements will participate in auto placement which is not intended.

**Comment by mattmcclinch** [comment]
So it needs to be set to false, which is what I have proposed.

**Comment by anatoly-os** [comment]
Ah, sorry. I misunderstood the sentense. Jetlag makes me a bit crazy, sorry :(
